lib/graphql/schema.rb

```diff
@@ -349,13 +349,9 @@ def lazy?(obj)
       !!lazy_method_name(obj)
     end
 
-    # @return [Array<#call>] Middlewares suitable for MiddlewareChain, applied to fields during execution
+    # @return [MiddlewareChain] MiddlewareChain which is applied to fields during execution
     def middleware
-      @middleware ||= if @rescue_middleware
-        [@rescue_middleware].concat(@user_middleware)
-      else
-        @user_middleware
-      end
+      @user_middleware
```

Comment:
Ah, that test was only good enough when it was a plain array. I added another spec in 0bb022b6c736c6f01ccf2f3199b1e005c1455b24